commit 09582bac29264997d71fcfb897d045c2dcac72f6
Author: Tim Duesterhus <tim@bastelstu.be>
Date:   Sat Jan 18 01:46:18 2020 +0100

    BUG/MINOR: cache: Fix leak of cache name in error path
    
    This issue was introduced in commit 99a17a2d91f9044ea20bba6617048488aed80555
    which first appeared in tag v1.9-dev11. This bugfix should be backported
    to HAProxy 1.9+.
    
    (cherry picked from commit d34b1ce5a20ce8f62b234f9696a621aaebe694c1)
    Signed-off-by: Christopher Faulet <cfaulet@haproxy.com>

diff --git a/src/cache.c b/src/cache.c
index 8e2acd1cb..dc11cf532 100644
--- a/src/cache.c
+++ b/src/cache.c
@@ -1452,7 +1452,7 @@ parse_cache_flt(char **args, int *cur_arg, struct proxy *px,
 			cconf = NULL;
 			memprintf(err, "%s: multiple explicit declarations of the cache filter '%s'",
 				  px->id, name);
-			return -1;
+			goto error;
 		}
 
 		/* Remove the implicit filter. <cconf> is kept for the explicit one */
